lti_consumer/models.py

```diff
@@ -487,8 +500,13 @@ def _get_lti_1p3_consumer(self):
         Uses the `config_store` variable to determine where to
         look for the configuration and instance the class.
         """
+        consumer_class = LtiAdvantageConsumer
+        # LTI Proctoring Services is not currently supported for CONFIG_ON_XBLOCK or CONFIG_EXTERNAL.
+        if self.lti_1p3_proctoring_enabled and self.config_store == self.CONFIG_ON_DB:
+            consumer_class = LtiProctoringConsumer
```

Comment:
@MichaelRoytman These lines mean that if `lti_1p3_proctoring_enabled` is true, then none of the advantage services will be available to that consumer, right? If we're working with a globally re-usable configuration then this adds a constraint: proctoring configs and other types of LTI configs must always be separate.

We don't have a use case for this right now, so this shouldn't block this ticket, but can you add a comment mentioning this new constraint with some context?

Reply:
_I have one idea in mind_: instead of using inheritance in the `LtiConsumer`, we could use composition, and add the services incrementally to the main class only if they are enabled in the configuration.